### DISCUSSION TITLE
ams OSRAM IAQ-CORE C - Unplausible CO₂ measurement readings

### DISCUSSION THREAD

**Discussion by Boldfor** [arendst/Tasmota#17563] [Q&A]
Based on @blakadder 's [idea](https://blakadder.com/vindriktning-tasmota/) I **integrated an [ams OSRAM IAQ-Core C indoor Air Quality Sensor Module]**(https://www.sciosense.com/wp-content/uploads/documents/iaQ-Core-Datasheet.pdf) into an [**IKEA Vindriktning air quality sensor**](https://www.ikea.com/de/de/p/vindriktning-luftqualitaetssensor-70498242/) with an **ESP-01S** according to the [information provided](https://tasmota.github.io/docs/IAQ/).

The hardware- and software part of this work, ..., except that the **sensor readings are off.** Not entirely off (more on this below), but just somehow off.

To make sure that I'm not missing anything I did replace the ESP itself, used different versions of Tasmota, and even bought a second IAQ-Core C sensor, to make sure that I'm not running a faulty sensor. Unfortunately, I always the same results:

- While the sensor always **hits the bottom of the spec (~450 ppm)**...
- ... the **spikes (>x.000 ppm) clearly don't make sense**, especially for such a short period of time
![image](https://user-images.githubusercontent.com/37384006/210156448-03138f32-6bc1-4e5d-a8ca-45ebb11f40a2.png)
- It's worth noting that the **spikes go in both directions** (resistance and TVOC for reference in that graph).
![image](https://user-images.githubusercontent.com/37384006/210156466-f87029ef-de17-4c06-bc15-2bac296ceb4d.png)
- What can also be seen quite nicely: Even if you ignore the spikes, the **absolute CO2 levels are reported for longer periods of time above levels which would be very harmful** (and I can safely say that in my flat we haven't had 3.000 ppm ;)).
![image](https://user-images.githubusercontent.com/37384006/210156569-e6c8f8c5-4169-4251-84c3-7783d3d3caaf.png)

From what I understood, the sensor only measures CO2 indirectly (Resistance in Ohm --> TVOC --> CO2), and is not calibrated in the factory but "calibrates itself" (?) during the startup phase and during operation. At the same time, it is not cheap either (US$ 20 - 40 per piece). So I wonder: Is the sensor just a poor sensor that produces "so so accurate" results? If yes, **which CO2 sensor works better**?

**Update:** I just read online that the sensors that use TVOC to extrapolate CO2 and do not compensate for humidity or temperature (like the mentioned ams IAQ-Core C) give more like an indication (good explanation see [here in the German make magazine](https://www.heise.de/select/make/2020/5/2022015381334973804)). The reason is (short version) that TVOC contains many gases (among others: CO2), and thus (since things like cooking, cleaning, etc impact TVOC but not necessarily CO2) calculating back to CO2 is not reliably possible. Some more explanations [here](https://wolles-elektronikkiste.de/tvoc-und-eco2-sensoren) (unfortunately also in German). If a TVOC-sensor is taken (instead of a CO2 sensor) to measure air quality, the WHO [recommends a threshold of < 0,30 mg/m³](https://www.catsensors.com/media/pdf/Sensor_Sensirion_IAM.pdf) (roughly translated into ~65 ppb).

**Update 2:** If CO2 is to be measured, the [MH-Z14 or MH-Z19](https://wolles-elektronikkiste.de/mh-z14-und-mh-z19-co2-sensoren) are somewhat better, and that the [SCD30](https://arnowelzel.de/ikea-vindriktning-mit-tasmota-und-scd30) is the best (and most expensive). Also good are the [sensors from Senseair](https://senseair.com/products?q=1331), but there's an [open bug](https://github.com/arendst/Tasmota/issues/16720) in Tasmota.
There‘s a good comparison of the various sensors [here](https://www.letscontrolit.com/forum/viewtopic.php?p=21691#p21691) and [here](http://www.ayt.cl/files/2019/Airthinx-IIT-Study.pdf).

**Comment by blakadder** [comment]
SCD41 is even better

**Reply by Boldfor** [reply]
Thanks!

Didn’t know that an NDIR is even possible in such a small form factor than the SCD41 (both the S8 as well as SCD30 look comparably big).

**Comment by Jason2866** [comment]
iaQ core is not an CO2 sensors. Do not expect correct CO2 values. It is a general Air Quality sensor. Whatever this means...

**Reply by Boldfor** [reply]
There's a pretty good explanation on what TVOC means [here](https://www.catsensors.com/media/pdf/Sensor_Sensirion_IAM.pdf).

The way I read it: Measuring both CO2 and TVOC makes sense. However, extrapolating from TVOC to CO2 (if CO2 is to be measured) does not make sense.

**Reply by blakadder** [reply]
It seems you've missed the first line in the datasheet:
"The iAQ-Core sensor module is used to measure VOC levels and provide CO2 equivalent and TVOC equivalent predictions"

focus on **equivalent** and **predictions**

**Reply by Boldfor** [reply]
Yes, you're right, I missed that point at the beginning.

Though, even after some reading, I still have problems understanding how the ams IAQ Core C can "measure VOC levels" but only "provide [...] TVOC equivalents". The only way this sentence can make sense is that _some_ VOCs are measured by the sensor, and then extrapolated to what's missing to the TVOC value.

Which leads to the question how accurate the "TVOC equivalents" are, given that the "CO2 equivalents" are not that precise.

**Reply by blakadder** [reply]
That would be a question for the manufacturer, Tasmota just reads the provided data via I2C

**Reply by Boldfor** [reply]
Yes, of course.

What I am still somewhat puzzled about: How a rather expensive (> 20 €) sensor of a European manufacturer (AMS) leaves some of these questions open. Both the S8 and the SCD4x appear to be way better documented (but then, to be fair, they're measuring just one discreet value (CO2), rather than VOC/TVOC, which does not seem to be clearly defined / a mixture of multiple gases in the air).

**Reply by sfromis** [reply]
Devices claiming to measure VOC are often **worse** for providing CO₂ data. The thing is that they "typically" base it all on a glob of corroded tin (also called tin oxide), that is entirely unable to identify any specific things in the air. The "sensor" simply reacts to changing resistance of the tin oxide. While CO₂ is one of the thing making a difference, many other things do too. The sensor can tell "something is in the air", the rest is speculative guesswork about what it is.

If you want reasonable CO₂ data, you need a dedicated sensor. SCD41 is a decent one.

**Reply by Boldfor** [reply]
Thanks. Will go for this one, as also recommended by @blakadder.

On a tangent: Do you have a recommendation for a reasonably-priced particle sensor that's supported by Tasmota? The Vindriktning is nice to get things going, but I read of high variances which leads nowhere on the long run.

**Reply by blakadder** [reply]
> Thanks. Will go for this one, as also recommended by @blakadder.
> 
> On a tangent: Do you have a recommendation for a reasonably-priced particle sensor that's supported by Tasmota? The Vindriktning is nice to get things going, but I read of high variances which leads nowhere on the long run.

PMSx003, vindriktning has the most basic version

**Reply by sfromis** [reply]
I have no comparisons of price or quality for these PM sensors supported (in addition to the Ikea one, which may have one of these):
```
Plantower PMS3003/5003/7003
Nova Fitness SDS011
Honeywell HPMA115S0
HM330X
SPS30
```
In general, expect sensors based on pulling air through the device (they basically need to do that) are likely to get dirty on the inside, much depending on the environment.
